canonical-web-and-design/maas-ui
# feat(ui): add ui key to Redux store

## Done

- Added `ui` key to redux, which for now will just house data regarding the header form.
- Updated some types to use `as const` for a bit more flexibility in how we use them.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open Redux DevTools and check there is a `ui` key (which you can't do anything with yet)
- Go to the machine list and check that you can still open action forms
- Check the same from a machine details page
- Go to a KVM details page and check that you can still open KVM forms
- Check that you can open machine forms from the VM table

Fixes #3023
